src/content/posts/docker_volume_introduction.md

```diff
@@ -0,0 +1,84 @@
+---
+title: "Use Docker's volumes to build a persistent log system"
+summary: "Persist data after container's death or share data accross containers through Docker's volume feature"
+date: 2023-09-19T12:00:00+0200
+lastUpdate: 2024-04-12T20:00:00+0200
+tags: ["docker", "volume", "log"]
+author: FireFrozen
+draft: false
+---
+
+Before reading this article, it's recommended to have a minimum knowledge of `docker`.  
+See, for example, the previous blog on docker: [Get Started with Docker](https://iscsc.fr/posts/short-docker-introduction/) by `ctmbl`.
+
+> **Disclaimer**:  
+> Note that in Docker, **two ways exist to persist/share data** with the host: **volumes** and **bind mounts**.  
+> While some key differences exist between both, they are not relevant in most of our basic use cases.  
+> We'll then **not distinguish one from the other**, some of what we say might better applied to bind mounts but **we'll only speak of volumes**.  
+> For more information see [Manage data in Docker](https://docs.docker.com/storage/).
+
+## What is a Docker's volume?
+
+A docker's volume is the action of **mounting an existing file or directory** of the host (the computer running the container) **into a container**. To rephrase, it enables a container to **share a memory space with the host**. So both the container and the user can access and modify it in real time and because it's not "contained" in the container but only mounted, it'll **persist after the container's death**.
+
+## Why do we need volumes?
+
+Docker containerization is very useful on lots of points for development but it's very annoying if your application **needs to produce persistent data**, like logs. Indeed if you **kill** your container (to update it for example) or if it **crashes**, you **lose everything that was inside** and understanding why your application crashed without its logs can be very complicated or even impossible.
+
+So we need to find a way to **keep certain files after the container's death** and that can be **achieved with volumes**!.
+
+## How to create a volume?
+
+To create a volume, you need to add a specific argument to the `docker run` command.
+
+> You can also create volumes outside of containers with [`docker volume`](https://docs.docker.com/storage/volumes/#create-and-manage-volumes)
+
+There is two different arguments which are really similar in their use. Those are `--mount` and `--volume`, both can be used to create volumes (or bind mounts, see Disclaimer above).  
```

Comment:
I think you should wipe the `--mount` part
As you said in the disclaimer, `--volume` is more appropriate for your usage, so it's a bit confusing that you still talk about it here.

Reply:
what I didn't understand before was that using `-v` doesn't necessarily creates volumes and using `--mount` doesn't necessarily creates bind mounts
see the TLDR of my answer below